source/1_engine/rex_engine/include/rex_engine/string/stringids.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "rex_std/string.h"
+
+namespace rex
+{
+  enum class SID
```

Comment:
not sure what this is for? is this ever needed?

Reply:
Yes to initialize a StringID to 0 but I can make it a state value does not have to be an enum

Reply:
why is a default construced StringID not initialized with 0?

Reply:
well I'd like to check it with neither really.
why don't you have a "empty()" func just like a regular string?

Reply:
For the same reason I like to check my booleans with == false and == true

Reply:
It's more readably IMO

Reply:
I can make it some constant does not have to be an enum, this was only there if there would be any other hardcoded values

Reply:
what is it actually that you want to test? if the StringID exists?
StringIDs should have the same interface as actual strings, and all string like interfaces already have "empty" func in there to check if there's actually something in it, now you can have a StringID that is empty so I suggest to use an interface func like "is_valid" or something.

Reply:
hardcoded values shouldn't be StringIDs, why can't they be string views?

Reply:
A StringID will never be empty, as it would store 0 or some other value, 0 is also a valid input so is_valid will also not really work. I'm not sure why StringID should have the same API as a string it's not a string. it's a hashed value to a string. For instance, you cannot query the size without looking up the actual string, you cannot iterate over it as with a string. That's why I would like to explicitly check if the value within the StringID == None as with empty or is_valid you don't really know what value is in it. You already have to know the API internally to know that empty or is_valid means None. StringID's do not store string_views they store ints and to compare a StringID with a string_view a lookup has to be made within the stringpool in order to check if it's similar which is where you will compare 2 strings instead of 2 integers if you compare with another StringID ( hardcoded or not )

Reply:
I already have an "is_none" function though, this checks on an empty SID
  //-------------------------------------------------------------------------
  /** True for StringID(), StringID(NAME_None) and StringID("None") */
  bool StringID::is_none() const
  {
    return *this == SID::None;
  }